scripts/cdc_prams/process.py

```diff
@@ -0,0 +1,758 @@
+
+    df['newStatVar'] = df['main_header'] + "_" + df[
+        'sub_header'] + "_" + df['statVar']
+    df = df.loc[(df['main_header_delete_flag'] == '')]
+    df = df.loc[(df['sub_header_delete_flag'] == '')]
+    df = df.drop(columns=[
+        'statVar', 'main_header_delete_flag', 'sub_header', 'main_header',
+        'sub_header_delete_flag'
+    ])
+    df['SV'] = df['newStatVar']
+    # Replacing statvar with proper statvar from the dictionary
+    df = df.replace({'SV': statvar_col})
+    # Resolving geoId using util folder.
+    df = df.replace({'Geo': _PLACE_MAP})
+    df = df.reset_index(drop=True)
+    # Creating a new column named as 'ScalingFactor'
+    df.insert(1, 'ScalingFactor', np.NaN)
+    return df
+
+
+def _splitting_ci_columns(df, flag):
+    '''
+    The CI has percent, lower confidence and upper confidence values
+    within one column. This method is used to seperated into three
+    different columns for State. Ex: 39.3 (36.4-42.2)
+    Args: DataFrame, Flag
+    Returns: df: DataFrame
+    '''
+    if flag == "States":
+        split_col = ['2016_CI', '2017_CI', '2018_CI', '2019_CI', '2020_CI']
+        for i in split_col:
+            df[i] = df[i].fillna(pd.NA)
+            # Splitting the column based on space and "-"
+            df_split = df[i].str.split(r"\s+|-", expand=True)
+            # determinign the size of the column after splitting it.
+            siz = df_split.shape[1]
+            # If the column is empty the size is 1 it is spit and
+            # the columns remain empty
+            if siz == 1:
+                df_split = df_split.rename(
+                    columns={df_split.columns[0]: i + '_PERCENT'})
+                df_split[i + '_LOWER'] = ""
+                df_split[i + '_UPPER'] = ""
+            # If the column size is 3 and hence it is split
+            elif siz == 3:
+                df_split = df_split.rename(
+                    columns={
+                        df_split.columns[0]: i + '_PERCENT',
+                        df_split.columns[1]: i + '_LOWER',
+                        df_split.columns[2]: i + '_UPPER'
+                    })
+                # Removing unwanted characters.
+                df_split[i + '_LOWER'] = df_split[i + '_LOWER'].str.replace(
+                    '(', '', regex=False)
+                df_split[i + '_UPPER'] = df_split[i + '_UPPER'].str.replace(
+                    ')', '', regex=False)
+            df = pd.concat([df, df_split], axis=1)
+        df = df.drop(columns=[
+            'newStatVar', '2016_CI', '2017_CI', '2018_CI', '2019_CI', '2020_CI',
+            'Overall_2020_CI'
+        ])
+        # Redifining columns
+    if flag == "National":
+        df = df[[
+            'Geo', 'SV', '2016_sampleSize', '2016_CI', '2017_sampleSize',
+            '2017_CI', '2018_sampleSize', '2018_CI', '2019_sampleSize',
+            '2019_CI', '2020_sampleSize', '2020_CI', 'ScalingFactor'
+        ]]
```

Comment:
generate dynamically instead of hardcoding

Reply:
Made it dynamic.